src/realm/object-store/impl/deep_change_checker.cpp

```diff
@@ -56,24 +55,35 @@ void DeepChangeChecker::find_related_tables(std::vector<RelatedTable>& related_t
     // so we rely on the fact that if there are any TypedLinks from a
     // Mixed value, there will be a corresponding backlink column
     // created at the destination table.
-    std::unordered_map<TableKey, LinkInfo> complete_mapping;
     Group* group = table.get_parent_group();
     REALM_ASSERT(group);
+    std::vector<LinkInfo> complete_mapping;
+    complete_mapping.resize(group->size());
+    auto get_mapping = [&](TableKey key) -> LinkInfo& {
+        size_t ndx = Group::key2ndx(key);
```

Comment:
It looks like table keys will be recycled after 100 table removals. I think that would lead to an already initialized LinkInfo being returned from here. Even if not, it seems like a risk that we are using implementation details of a TableKey (though we have done worse in the name of performance). Could we get similar performance improvement by using our `util::FlatMap`?

Reply:
This is a temporary data structure that doesn't outlive this function, and the state of the transaction it's reading from can't change while it exists.

Reply:
Oh, you're right. All good then, thanks! 👍